src/services/review/RepoCheckerService.ts

```diff
@@ -64,6 +69,65 @@ export default class RepoCheckerService {
     this.repoRepository = repoRepository
   }
 
+  isCurrentlyLocked(repo: string): ResultAsync<boolean, never> {
+    const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
+    // create logs folder if it does not exist
+
+    if (!fs.existsSync(logsFilePath)) {
+      fs.mkdirSync(logsFilePath, { recursive: true })
+    }
+
+    // check if checker.lock exists
+    const lockFilePath = path.join(logsFilePath, "checker.lock")
+    return ResultAsync.fromPromise(
+      fs.promises.access(lockFilePath, fs.constants.F_OK),
+      () => okAsync(false)
+    )
+      .map(() => true)
+      .orElse(() => ok(false))
+  }
```

Comment:
we could make this better by returning an `error` that describes the failure mode instead of just `false`

Reply:
Hmm may i clarify if this is a nit? 

from the pov of the caller, feels like `ResultAsync<boolean, never>` is better contract than `ResultAsync<true, false>`, since the latter needs to do a  `.orElse(ok(false))`. 
ie an error thrown here is not a sign of something went wrong in trying to get the result of "is this repo locked" (which signals intent of some error recovery is needed), but rather just the repo not locked at the moment

Reply:
this is a nit in the context of this PR (get this out to proto); `ResultAsync` contains 2 types in itself - we want to return `ResultAsync<true, someError>` because the error itself is more descriptive that `false`. taken in context, this means that: 

1. if i return `ResultAsync<success, false>` - i know only about the presence of an error, but idk _what_ error occurred. 
2. if i return `ResultAsync<boolean, never>` - `neverthrow` isn't adding much value here. the type signature indicates that this doesn't throw so why not just return `boolean`? (but then same point as above) 

we don't necessarily have to fix this _now_ cos this is a proto